presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanAssert.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.assertions;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.TableMetadata;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.JoinNode;
+import com.facebook.presto.sql.planner.plan.JoinNode.EquiJoinClause;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.SemiJoinNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+import com.facebook.presto.sql.tree.Expression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.planner.PlanPrinter.textLogicalPlan;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertTrue;
+
+public class PlanAssert
+{
+    private PlanAssert() {}
+
+    public static void assertPlan(Session session, Metadata metadata, Plan actualPlan, PlanMatcher expectedPlanMatcher)
+    {
+        requireNonNull(actualPlan, "root is null");
+
+        boolean matches = actualPlan.getRoot().accept(new PlanMatchingVisitor(session, metadata), new PlanMatchingContext(expectedPlanMatcher));
+        if (!matches) {
+            String logicalPlan = textLogicalPlan(actualPlan.getRoot(), actualPlan.getTypes(), metadata, session);
+            assertTrue(matches, format("Plan does not match:\n %s", logicalPlan));
+        }
+    }
+
+    public static PlanMatcher node(Class<? extends PlanNode> nodeClass, PlanMatcher... sources)
+    {
+        return anyNode(sources).with((node, session, metadata, symbolAliases) -> node.getClass().equals(nodeClass));
+    }
+
+    public static PlanMatcher anyNode(PlanMatcher... sources)
```

Comment:
Why not call this just "any", the one below "anyTree", and drop the "Node" from all the remaining ones: "tableScan", "semiJoin", "project", etc.